### PR TITLE
Get Sphinx build working in CI

### PR DESCRIPTION
**Pull request by jessica-mitchell**
This  PR updates the nestbuildmatrix for sphinx-rtd so that it builds Sphinx.

### CODE REVIEW

File: .github/workflows/nestbuildmatrix.yml [hunk 1]
```diff
@@ -335,22 +335,24 @@ jobs:
       - name: "Set up Python 3.x"
         uses: actions/setup-python@v4
         with:
-          python-version: 3.9
+          python-version: 3.8
```
Comment (terhorstd) [inline .github/workflows/nestbuildmatrix.yml:338]: What is the critical part depending on the old version?

**Review by terhorstd — CHANGES_REQUESTED** [review]
Hey! Thanks for the fixes! – Maybe add some comments with a bit of context, so that the next person looking at this understands the motivation.

**Comment by jessica-mitchell** [comment]
@terhorstd I added some comments with links to better explain the install for pandoc and specifying the Python version

File: .github/workflows/nestbuildmatrix.yml (hunk 1 above)
Comment (terhorstd) [reply .github/workflows/nestbuildmatrix.yml:340]: Thanks for adding the comment.

File: .github/workflows/nestbuildmatrix.yml [hunk 2]
```diff
@@ -335,22 +335,24 @@ jobs:
...
 
       - name: "Install dependencies"
         run: |
           pip install -r doc/requirements.txt
+          sudo apt-get install pandoc
```
[the remark this answers is not in the document]
Comment (jessica-mitchell) [reply .github/workflows/nestbuildmatrix.yml:346]: I added a comment (See my change) , pip does not lead to broken version, from what I understand is the 'executable' part is not pip installable

**Review by terhorstd — APPROVED** [review]
LGTM! :)

**Pull request merged**